### DISCUSSION TITLE
Running Argo with applications in any namespace

### DISCUSSION THREAD

**Discussion by r-bennett** [argoproj/argo-cd#15308] [Q&A]
I am trying to set up a cluster-scoped Argo instance with the [applications in any namespace](https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/) feature. This is so that we can have a centrally managed instance of Argo, with centrally managed projects and app-of-apps, but then allow users to self-service their own applications.

So I have something like:
```
- k8s cluster
  - control_plane_namespace
    - <argo deployment>
    - <argo CRDs for projects and app-of-apps>
  - user_controlled_namespace_1
    - <self-service argo CRDS for apps>  
  - user_controlled_namespace_2
    - <self-service argo CRDS for apps>
```

This is all OK and my self service apps are picked up and synced. 

However, I also have other cluster-scoped Argo instances in other namespaces on the same cluster (these are not using applications in any namespace).

I have set the `ARGOCD_APPLICATION_NAMESPACES` env var to `control_plane_namespace,user_controlled_namespace_1,user_controlled_namespace_2` for both `argocd-application-controller` and `argocd-server`, which as I understand it means that my new argo instance should only process application CRDs from those namespaces, which is what I want. I can confirm that this is being applied correctly from the logs.

But the new Argo instance still picks up application CRDs from all other namespaces in the same cluster, despite the fact that they are outside the list of namespaces specified. (The apps it picks up are from the `default` project, which exists on all Argos).

Any idea why we're still picking things up from other namespaces, or how I can debug further?

**Comment by jannfis** [comment]
Hi. What exactly do you mean by "application is picked up"? Can you please be a little more concise?

**Reply by jannfis** [reply]
Also, which version of Argo CD are you running?

**Reply by r-bennett** [reply]
This looks to be https://github.com/argoproj/argo-cd/issues/13077, I see a fix is incoming 🥳
